networkit/cpp/community/OverlappingNMIDistance.cpp

```diff
@@ -0,0 +1,393 @@
+    if (value > upper_bound) {
+        if (!Aux::NumericTools::le(value, upper_bound, Aux::NumericTools::acceptableError)) {
+            std::stringstream ss;
+            ss << std::fixed << std::setprecision(print_precision) << value;
+            const auto s = ss.str();
+            ERRORF(format, s.c_str());
+        }
+        value = upper_bound;
+    }
+}
+
+/**
+ * Normalize the given mutual information value.
+ *
+ * @param normalization
+ * @param mutual_information
+ * @param entropy_X
+ * @param entropy_Y
+ * @return
+ */
+double normalize(OverlappingNMIDistance::Normalization normalization, double mutual_information,
+                 double entropy_X, double entropy_Y) {
+    // Numerical errors less than Aux::NumericTools::acceptableError are corrected silently.
+    // Larger errors are also corrected, but they are logged as an ERROR.
+    clamp_below(entropy_X, 0.0, "Set entropy_X lower than 0.0 to 0.0: %s");
+    clamp_below(entropy_Y, 0.0, "Set entropy_Y lower than 0.0 to 0.0: %s");
+
+    // Assumption: Empty covers were already dealt with.
+    // At this point, H(X) = 0 implies that X has only clusters with all nodes.
+    if (entropy_X == 0.0 && entropy_Y == 0.0) {
+        // Duplicate clusters should not exist, but its not explicitly checked.
+        // De-duplicating clusters results in two covers that are equal.
+        return 1.0;
+    } else if ((entropy_X == 0.0 || entropy_Y == 0.0)
+               && (normalization == OverlappingNMIDistance::Normalization::MIN
+                   || normalization == OverlappingNMIDistance::Normalization::GEOMETRIC_MEAN)) {
+        // Prevent division by zero.
+        // One of the covers has only clusters with all nodes. The other cover is different.
+        return 0.0;
+    }
+
+    double nmi;
+    switch (normalization) {
+    case OverlappingNMIDistance::Normalization::MIN:
+        nmi = mutual_information / std::min(entropy_X, entropy_Y);
+        break;
+    case OverlappingNMIDistance::Normalization::GEOMETRIC_MEAN:
+        nmi = mutual_information / std::sqrt(entropy_X * entropy_Y);
+        break;
+    case OverlappingNMIDistance::Normalization::ARITHMETIC_MEAN:
+        nmi = 2 * mutual_information / (entropy_X + entropy_Y);
+        break;
+    case OverlappingNMIDistance::Normalization::MAX:
+        nmi = mutual_information / std::max(entropy_X, entropy_Y);
+        break;
+    case OverlappingNMIDistance::Normalization::JOINT_ENTROPY: {
+        auto entropy_XY = entropy_X + entropy_Y - mutual_information;
+        nmi = mutual_information / entropy_XY;
+        break;
+    }
+    default:
+        throw std::logic_error("normalization method is not covered");
+    }
+
+    if (std::isnan(nmi)) {
+        ERROR("Set nmi ", nmi, " to 0.0");
+        nmi = 0.0;
+    }
+    clamp_below(nmi, 0.0, "Set nmi lower than 0.0 to 0.0: %s");
+    clamp_above(nmi, 1.0, "Set nmi larger than 1.0 to 1.0: %s");
+
+    return nmi;
+}
+} // namespace
```

Comment:
I think it's a better idea to implement these utility methods as private static members of `OverlappingNMIDistance`.
Also, please use `camelBack`.

Reply:
Done (4685253 and 18f2d4d)